### ISSUE TITLE
Add Telegraf support for VictoriaLogs

### ISSUE THREAD

**Issue by denisgolius** [VictoriaMetrics/VictoriaMetrics#6244]
### Is your feature request related to a problem? Please describe

Telegraf supports parsing logs and sends them to Elasticsearch. VictoriaLogs do support some Elasticsearch API, so I was trying to write the data from Telegraf to the VictoriaLogs, but I've failed in issue because Telegraf wants to write to the specific path: `_template/telegraf`, see a log below:

```bash
vlogs-1  | 2024-04-30T15:55:47.510Z     warn    VictoriaMetrics/lib/httpserver/httpserver.go:419        remoteAddr: "192.168.1.5:38368"; requestURI: /insert/elasticsearch/_template/telegraf; unsupported path requested: "/insert/elasticsearch/_template/telegraf"
```

### Describe the solution you'd like

It would be great to add support to write the logs data from Telegraf by adding a new route on the VictoriaLogs side.

### Additional information

I used:
- Ubuntu 22.04
- telegraf from [Influxdata repository for ubuntu](https://www.influxdata.com/downloads/) 
- Telegraf configuration example for Elasticsearch https://github.com/influxdata/telegraf/issues/5676
- VictoriaLogs in docker from https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/deployment/docker/victorialogs/fluentbit-docker/docker-compose.yml#L13-L23

```bash
telegraf --version
Telegraf 1.30.2 (git: HEAD@10c15ce4)
```

Telegraf agent configuration example from `/etc/telegraf/telegraf.conf`:
```bash
[agent]
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 100000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "0s"
  precision = ""
  debug = false
  quiet = false
  logtarget = "file"
  logfile = "/dev/null"
  hostname = "pop-os"
  omit_hostname = false

[[outputs.elasticsearch]]
  urls = ["http://192.168.1.5:9428/insert/elasticsearch"]
  timeout = "1m"
  flush_interval = "30s"
  enable_sniffer = false
  health_check_interval = "0s"
  index_name = "device_log-%Y.%m.%d"
  manage_template = true
  template_name = "telegraf"
  overwrite_template = false
  namepass = ["tail"]

# # Parse the new lines appended to a file
[[inputs.tail]]
  files = ["/var/log/telegraf/telegraf.log"]
  from_beginning = false
  interval = "10s"
  pipe = false
  watch_method = "inotify"
  data_format = "value"
  data_type = "string"
  character_encoding = "utf-8"

  [inputs.tail.tags]
     metric_type = "logs"
     log_source = "telegraf"
```

**Comment by AndrewChubatiuk** [comment]
disabled `manage_template` to get rid of this error, telegraf is able to push data to victorialogs, but there's no way how to remap it's fields to _msg_fields, _stream_fields, _time_fields. It's possible to add PR to telegraf to add custom params on ingestion

**Comment by valyala** [comment]
@AndrewChubatiuk , it is better from usability perspective to handle all the HTTP requests starting from `insert/elasticsearch/_template/` in a generic way at VictoriaLogs side, so client won't notice these paths are ignored.

As for the support of [additional http parameters](https://docs.victoriametrics.com/victorialogs/data-ingestion/#http-parameters) at Telegraf configs, could you create a pull request at [Telegraf GitHub repository](https://github.com/influxdata/telegraf), which adds an ability to configure any http parameters in Telegraf configs?

While these params aren't supported by Telegraf, it would be great figuring out and using the best values for these parameters at VictoriaLogs side when it detects logs received from Telegraf in Elasticsearch format.

**Comment by AndrewChubatiuk** [comment]
@valyala main issue is with library, which telegraf is using for elasticsearch requests. It bulk request doesn't not support custom query args and the latest commit was about two years ago https://github.com/olivere/elastic/blob/29ee98974cf1984dfecf53ef772d721fb97cb0b9/bulk.go#L233

**Comment by AndrewChubatiuk** [comment]
custom headers are supported by telegraf starting from [release 1.31.0](https://github.com/influxdata/telegraf/releases/tag/v1.31.0)

**Comment by AndrewChubatiuk** [comment]
@valyala custom headers are supported by telegraf starting from [release 1.31.0](https://github.com/influxdata/telegraf/releases/tag/v1.31.0)

**Comment by hagen1778** [comment]
@AndrewChubatiuk would you mind adding a PR with example of configuring telegraf to push logs to VictoriaLogs, similar to other examples https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/app/vlinsert ?

**Comment by AndrewChubatiuk** [comment]
it's already [there](https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/deployment/docker/victorialogs/telegraf-docker)

**Comment by AndrewChubatiuk** [comment]
telegraf logs support was included into VictoriaLogs release [0.29.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v0.29.0-victorialogs). Please check how to ingest logs from Telegraf [here](https://github.com/VictoriaMetrics/VictoriaMetrics/tree/master/deployment/docker/victorialogs/telegraf)
